nixos/modules/services/web-apps/nextcloud-notify_push.nix

```diff
@@ -0,0 +1,98 @@
+{ config, options, lib, pkgs, ... }:
+
+let
+  cfg = config.services.nextcloud.notify_push;
+in
+{
+  options.services.nextcloud.notify_push = {
+    enable = lib.mkEnableOption "Notify push";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.nextcloud-notify_push;
+      defaultText = lib.literalMD "pkgs.nextcloud-notify_push";
+      description = "Which package to use for notify_push";
+    };
+
+    socketPath = lib.mkOption {
+      type = lib.types.str;
+      default = "/run/nextcloud-notify_push/sock";
+      description = "Socket path to use for notify_push";
+    };
+
+    logLevel = lib.mkOption {
+      type = lib.types.enum [ "error" "warn" "info" "debug" "trace" ];
+      default = "error";
+      description = "Log level";
+    };
+  } // (
+    lib.listToAttrs (
+      map (
+        opt: lib.nameValuePair opt (options.services.nextcloud.config.${opt} // {
```

Comment:
Wouldn't `lib.genAttrs` be more suitable (and also simpler) here? Or am I missing something? :sweat_smile:

Reply:
You're missing that I forgot about `lib.genAttrs` :grimacing: 

Added now.

This has become quite a few commits now… do you think I should squash some of them?

Reply:
We could squash all things matching ``nixos/nextcloud-notify_push*``